From 8d0f169854a2a33256b8a2c7413d869bf8b46f8c Mon Sep 17 00:00:00 2001
From: b02247 <b02247@freescale.com>
Date: Thu, 1 Nov 2012 10:19:56 +0800
Subject: [PATCH 01/25] ENGR00231966 add usb mic support in jb

1.add usb mic support in audio policy manager

Change-Id: I43e33ffad66309d29fd30283a8e6a8446054672f
Signed-off-by: b02247 <b02247@freescale.com>

Conflicts:
	audio/AudioPolicyManagerBase.cpp
---
 audio/AudioPolicyManagerBase.cpp | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/audio/AudioPolicyManagerBase.cpp b/audio/AudioPolicyManagerBase.cpp
index 85b0084..abfd143 100644
--- a/audio/AudioPolicyManagerBase.cpp
+++ b/audio/AudioPolicyManagerBase.cpp
@@ -1,5 +1,6 @@
 /*
  * Copyright (C) 2009 The Android Open Source Project
+ * Copyright (C) 2012 Freescale Semiconductor, Inc.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -2952,7 +2953,9 @@ audio_devices_t AudioPolicyManagerBase::getDeviceForInputSource(int inputSource)
         }
         break;
     case AUDIO_SOURCE_CAMCORDER:
-        if (mAvailableInputDevices & AUDIO_DEVICE_IN_BACK_MIC) {
+        if (mAvailableInputDevices & AUDIO_DEVICE_IN_USB_DEVICE) {
+            device = AUDIO_DEVICE_IN_USB_DEVICE;
+        } else if (mAvailableInputDevices & AUDIO_DEVICE_IN_BACK_MIC) {
             device = AUDIO_DEVICE_IN_BACK_MIC;
         } else if (mAvailableInputDevices & AUDIO_DEVICE_IN_BUILTIN_MIC) {
             device = AUDIO_DEVICE_IN_BUILTIN_MIC;
-- 
1.8.0

